packages/atomic/src/components/commerce/atomic-commerce-interface/analytics-config.ts

```diff
@@ -0,0 +1,23 @@
+import {
+  AnalyticsConfiguration,
+  CommerceEngineConfiguration,
+} from '@coveo/headless/commerce';
+
+export function getAnalyticsConfig(
+  commerceEngineConfig: CommerceEngineConfiguration,
+  enabled: boolean
+): AnalyticsConfiguration {
+  const defaultConfiguration: AnalyticsConfiguration = {
+    enabled,
+    documentLocation: document.location.href,
```

Comment:
Actually, it should be enforced, so in the spread, we should:
- have `...{analyticsMode: 'next'} last
- Enforce/check/marshall the presence of analytics.trackingId. Unless we want a default tracking ID (I'm not sure tho abt that.)

Reply:
About that, what should happen if trackingId is not set?

Reply:
The `trackingId` is mandatory for the construction of a commerce engine, and it should be mandatory for the atomic commerce interface IMO. We should error out if one isn't provided.

Reply:
> The `trackingId` is mandatory for the construction of a commerce engine, and it should be mandatory for the atomic commerce interface IMO. We should error out if one isn't provided.

The Commerce Engine will :tableflip: when we try to instantiate then :D
However, again, this is a cross-use-case concern, so it may be centralized/refactored in a single place (and I'd not consider it a blocking requirement for the merge of this PR

Reply:
Will be addressed in https://coveord.atlassian.net/browse/KIT-3137

Reply:
Just confirming that a `trackingId` is required 👍